packages/flutter_tools/test/general.shard/ios/ios_device_install_test.dart

```diff
@@ -76,28 +76,113 @@ void main() {
     expect(processManager.hasRemainingExpectations, false);
   });
 
-  testWithoutContext('IOSDevice.isAppInstalled catches ProcessException from ios-deploy', () async {
-    final IOSApp iosApp = PrebuiltIOSApp(projectBundleId: 'app');
-    final FakeProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
-      FakeCommand(command: const <String>[
-        'ios-deploy',
-        '--id',
-        '1234',
-        '--exists',
-        '--bundle_id',
-        'app',
-      ], environment: const <String, String>{
-        'PATH': '/usr/bin:null',
-        ...kDyLdLibEntry,
-      }, onRun: () {
-        throw const ProcessException('ios-deploy', <String>[]);
-      })
-    ]);
-    final IOSDevice device = setUpIOSDevice(processManager: processManager);
-    final bool isAppInstalled = await device.isAppInstalled(iosApp);
+  group('isAppInstalled', () {
+    BufferLogger logger;
```

Comment:
Is it really worth a setUp for a logger that is used in a single test? FWIW This will eventually become a `BufferLogger?` with null safety, meaning you'll need to assert `!` or null check to use it safely.

Reply:
It's used in two tests but I can push it down into the test.